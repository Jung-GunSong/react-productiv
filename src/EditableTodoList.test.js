import React from "react";
import { render, fireEvent } from "@testing-library/react";
import EditableTodoList from "./EditableTodoList";

const testTodos = [{
  id: 1,
  title: "test1",
  description: "testing1",
  priority: 2,
},
{
  id: 2,
  title: "test2",
  description: "testing testing",
  priority: 1,
},
{
  id: 3,
  title: "test3",
  description: "still testing....",
  priority: 3,
}];


describe("EditableTodoList component", function () {
  it("renders without crashing", function () {
    render(<EditableTodoList todos={testTodos} />);
  });

  it("matches snapshot", function () {
    const { container } = render(<EditableTodoList todos={testTodos} />);
    expect(container).toMatchSnapshot();
  });

  it("renders todos correctly", function () {
    const { queryByText, container } = render(<EditableTodoList todos={testTodos} />);
    expect(queryByText("test1")).toBeInTheDocument();
    expect(queryByText("testing1")).toBeInTheDocument();
    expect(container.querySelector(".EditableTodo-toggle")).toBeInTheDocument();
    expect(container.querySelector(".EditableTodo-delBtn")).toBeInTheDocument();
  });

  it("renders edit form on click", function () {
    const { container } = render(<EditableTodoList todos={testTodos} />);

    //click edit button
    fireEvent.click(container.querySelector(".EditableTodo-toggle"));
    expect(container.querySelector(".NewTodoForm")).toBeInTheDocument();
  });

});
