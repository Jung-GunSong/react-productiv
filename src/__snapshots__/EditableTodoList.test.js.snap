// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditableTodoList component matches snapshot 1`] = `
<div>
  <div>
    <div
      class="EditableTodo"
    >
      <div
        class="mb-3"
      >
        <div
          class="float-end text-sm-end"
        >
          <button
            class="EditableTodo-toggle btn-link btn btn-sm"
          >
            Edit
          </button>
          <button
            class="EditableTodo-delBtn btn-link btn btn-sm text-danger"
          >
            Del
          </button>
        </div>
        <div
          class="Todo"
        >
          <div>
            <b>
              test1
            </b>
             
            <small>
              priority: 
              2
            </small>
          </div>
          <div>
            <small>
              testing1
            </small>
          </div>
        </div>
      </div>
    </div>
    <div
      class="EditableTodo"
    >
      <div
        class="mb-3"
      >
        <div
          class="float-end text-sm-end"
        >
          <button
            class="EditableTodo-toggle btn-link btn btn-sm"
          >
            Edit
          </button>
          <button
            class="EditableTodo-delBtn btn-link btn btn-sm text-danger"
          >
            Del
          </button>
        </div>
        <div
          class="Todo"
        >
          <div>
            <b>
              test2
            </b>
             
            <small>
              priority: 
              1
            </small>
          </div>
          <div>
            <small>
              testing testing
            </small>
          </div>
        </div>
      </div>
    </div>
    <div
      class="EditableTodo"
    >
      <div
        class="mb-3"
      >
        <div
          class="float-end text-sm-end"
        >
          <button
            class="EditableTodo-toggle btn-link btn btn-sm"
          >
            Edit
          </button>
          <button
            class="EditableTodo-delBtn btn-link btn btn-sm text-danger"
          >
            Del
          </button>
        </div>
        <div
          class="Todo"
        >
          <div>
            <b>
              test3
            </b>
             
            <small>
              priority: 
              3
            </small>
          </div>
          <div>
            <small>
              still testing....
            </small>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
